src/graph/edge_extractor.py

```
@@ -0,0 +1,381 @@
+        Args:
+            file_data: Processed file data from FileProcessor
+            
+        Returns:
+            Dictionary of edge types to edge lists
+        """
+        edges = defaultdict(list)
+        
+        # Handle both dict and ProcessedFile inputs for compatibility
+        if hasattr(file_data, 'type'):
+            # ProcessedFile object
+            file_type = getattr(file_data, 'type')
```

Ruff (0.11.9)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)